common/app/com/commercetools/sunrise/common/httpauth/basic/BasicHttpAuthentication.java

```diff
@@ -0,0 +1,54 @@
+package com.commercetools.sunrise.common.httpauth.basic;
+
+import com.commercetools.sunrise.common.httpauth.HttpAuthentication;
+import io.sphere.sdk.models.Base;
+
+import javax.annotation.Nullable;
+import java.util.Base64;
+
+/**
+ * Enables HTTP Basic Access Authentication.
+ */
+final class BasicHttpAuthentication extends Base implements HttpAuthentication {
+
+    @Nullable
+    private final String realm;
+    @Nullable
+    private final String encodedCredentials;
+
+    BasicHttpAuthentication() {
+        this.realm = null;
+        this.encodedCredentials = null;
+    }
+
+    BasicHttpAuthentication(final String realm, final String credentials) {
+        this.realm = realm;
+        this.encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes());
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return encodedCredentials != null;
+    }
+
+    @Override
+    public String getWwwAuthenticateHeader() {
```

Comment:
maybe this could be `getWwwAuthenticateHeaderValue` since it is not returning the header including the name